### ISSUE TITLE
Improve Speed of HLS Manifest Generation

### ISSUE THREAD

**Issue by joncameron** [avalonmediasystem/avalon#5993]
# Description
HLS playlists are slow to return when requested by Ramp, slowing down the entire loading of the page.

In MCO, once the Ramp media player loads, it still takes about 4 seconds for Ramp to be ready to begin playing content. Is this slowness coming from the HLS manifest? For a given record on MCO, the m3u8 request is taking about 2.5 seconds, and seems to be the slowest request initiated when loading a media item.

# Done Looks Like
- [x] Look for requests that contribute to slowness of being able to load content into the Ramp player
- [x] Ideas for how we might speed up the HLS manifest generation?

**Comment by elynema** [comment]
Looks like something in the ability checks in Solr may be contributing. Masterfile controller checks to see if user can read the masterfile, but 8 Solr requests looking up the media object may be too many. Need to trace that and see if can figure out what's happening.

**Comment by masaball** [comment]
@cjcolvar and I paired up and were able to track down the root of the large amount of solr calls being seen. There is an inefficiency in SpeedyAF where certain methods are not caching their results. Even though we had already pulled the MediaObject from solr the proxy was not being retained and so each subsequent MediaObject call would re-query solr. There were ultimately a couple MediaObject calls occurring in the ability check workflow and a few in `MasterFile#stream_details`, as well as at least one Derivative call.

By explicitly loading the reflections when we initially populate the MasterFile proxy, we are able to force the caching of all related objects (MediaObject, Derivative, SupplementalFiles, etc). We also rearranged the method that initializes the MasterFile proxy so that solr is only queried once for that operation. This will improve the speed of every MasterFile controller action.

Break down of HLS manifest generation after these changes, timings are from mco-staging (prod should hopefully be faster since solr server is local for prod):
- 1 solr request to initialize the MasterFile proxy (200-300ms on average)
- 1 solr request to initialize the reflections (200-300ms on average)
- 1 solr request to check permissions (~50ms)
- 1 solr request for parent collection (200-300ms on average)
- 3 requests to wowza to grab the stream urls (150-300ms on average)

This cuts the number of requests at least in half, saving 1-2 seconds from the quick testing Chris and I did. 

The query for the collection was not affected by using `load_reflections`. We could potentially save that ~250ms by passing the media object proxy into `SecurityHelper#secure_streams` but we did not pursue that avenue at this time. There is unlikely to be much additional time past that which could be saved in Avalon itself.
